src/client/algod.js

```diff
@@ -21,17 +21,25 @@ function Algod(token, baseServer = "http://r2.algorand.network", port = 4180) {
      * @returns {Promise<*>}
      */
     this.status = async function () {
-        let res = await c.get("/v1/status");
-        return res.body;
+        try {
+            let res = await c.get("/v1/status");
+            return res.body;
+        } catch (e) {
+            throw Error(e.response.error.message)
+        }
     };
 
     /**
      * helathCheck returns an empty object iff the node is running
      * @returns {Promise<*>}
      */
     this.healthCheck = async function () {
-        let res = await c.get("/health");
-        return res.body;
+        try {
+            let res = await c.get("/health");
+            return res.body;
+        } catch (e) {
+            throw Error(e.response.error.message)
```

Comment:
You should throw back the entire error. You lose a lot of information by wrapping just the message

Reply:
@rotemh It turned out that the error itself was being propagated up just fine, it was not being swallowed; the ticket's problem was that the thrown error was not of the expected format. (I.e. in the example, if one catches the resulting error `e`, one can `console.log(e)`, but `console.log(e.message)` will be undefined.)

I'll look into a version that defines `e.message` while preserving the rest of the information. :+1: